packages/web-components/fast-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,324 @@
+            const prev: number = this.activeIndicatorRef.offsetLeft;
+            this.activeIndicatorRef.style.gridColumn = `${this.activeTabIndex + 1}`;
+            const next: number = this.activeIndicatorRef.offsetLeft;
+            this.activeIndicatorRef.style.gridColumn = `${this.prevActiveTabIndex + 1}`;
+            const dif: number = next - prev;
+            this.activeIndicatorRef.style.transform = `translateX(${dif}px)`;
+            this.activeIndicatorRef.classList.add("activeIndicatorTransition");
+            this.activeIndicatorRef.addEventListener("transitionend", () => {
+                this.ticking = false;
+                this.activeIndicatorRef.style.gridColumn = `${this.activeTabIndex + 1}`;
+                this.activeIndicatorRef.style.transform = "translateX(0px)";
+                this.activeIndicatorRef.classList.remove("activeIndicatorTransition");
+            });
+        } else {
+            const prev: number = this.activeIndicatorRef.offsetTop;
+            this.activeIndicatorRef.style.gridRow = `${this.activeTabIndex + 1}`;
+            const next: number = this.activeIndicatorRef.offsetTop;
+            this.activeIndicatorRef.style.gridRow = `${this.prevActiveTabIndex + 1}`;
+            const dif: number = next - prev;
+            this.activeIndicatorRef.style.transform = `translateY(${dif}px)`;
+            this.activeIndicatorRef.classList.add("activeIndicatorTransition");
+            this.activeIndicatorRef.addEventListener("transitionend", () => {
+                this.ticking = false;
+                this.activeIndicatorRef.style.gridRow = `${this.activeTabIndex + 1}`;
+                this.activeIndicatorRef.style.transform = "translateX(0px)";
+                this.activeIndicatorRef.classList.remove("activeIndicatorTransition");
+            });
+        }
+    }
+
+    private decrement(tabsLength: number): void {
+        this.prevActiveTabIndex = this.activeTabIndex;
+        if (this.activeTabIndex !== 0) {
+            this.activeTabIndex = this.activeTabIndex - 1;
+        } else {
+            this.activeTabIndex = tabsLength - 1;
+        }
+        this.setTabs();
+    }
+
+    private increment(tabsLength: number): void {
+        this.prevActiveTabIndex = this.activeTabIndex;
+        if (this.activeTabIndex !== tabsLength - 1) {
+            this.activeTabIndex = this.activeTabIndex + 1;
+        } else {
+            this.activeTabIndex = 0;
+        }
+        this.setTabs();
+    }
+
+    private focusTab(): void {
+        const tb = this.tabs as HTMLElement[];
+        tb[this.activeTabIndex].focus();
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        this.connected = true;
+    }
+
+    constructor() {
+        super();
+
+        if (this.connected) {
+            this.tabIDs = this.getTabIds();
+            this.tabPanelIDs = this.getTabPanelIds();
+            this.activeTabIndex = this.getActiveIndex();
+        }
```

Comment:
is this case ever true? It seems `connected` always gets asigned true in the connectedCallback which should always be called after the constructor.